src/pubkey.cpp

```diff
@@ -188,6 +188,18 @@ bool XOnlyPubKey::CheckPayToContract(const XOnlyPubKey& base, const uint256& has
     return secp256k1_xonly_pubkey_tweak_add_check(secp256k1_context_verify, m_keydata.begin(), parity, &base_point, hash.begin());
 }
 
+bool CPubKey::TweakMulVerify(const CPubKey& untweaked, const uint256& tweak) const
+{
+    assert(this->IsCompressed() && untweaked.IsCompressed());
+    secp256k1_pubkey pk;
+    if (!secp256k1_ec_pubkey_parse(secp256k1_context_verify, &pk, untweaked.data(), untweaked.size())) return false;
+    if (!secp256k1_ec_pubkey_tweak_mul(secp256k1_context_verify, &pk, tweak.data())) return false;
+    unsigned char out_pk[CPubKey::COMPRESSED_SIZE];
+    size_t out_len = CPubKey::COMPRESSED_SIZE;
+    if (!secp256k1_ec_pubkey_serialize(secp256k1_context_verify, out_pk, &out_len, &pk, SECP256K1_EC_COMPRESSED)) return false;
+    return *this == CPubKey(out_pk, out_pk + CPubKey::COMPRESSED_SIZE);
```

Comment:
Perhaps `assert(CPubKey::COMPRESSED_SIZE == out_len)`, or use `out_pk + out_len` instead of `out_pk + CPubKey::COMPRESSED_SIZE`.